vuln-reach/src/javascript/lang/symbol_table.rs

```diff
@@ -158,6 +180,16 @@ impl<'a> SymbolTableBuilder<'a> {
 
                 self.visit_children(node);
             },
+            "assignment_expression" | "augmented_assignment_expression" => {
+                // Add assignments to their scope, allowing identification of hoisted variables.
+                let scope = self.find_parent_function_scope().unwrap();
+                let name = node.child_by_field_name(b"left").unwrap();
+                if !scope.names.contains(&name) {
```

Comment:
This is a tiny optimization which should help with all well behaved JavaScript code.

So the only cases where things will be added is if it's either going to be hoisted, or if the declaration is after the assignment. Realistically both of these are something I'd consider poor JavaScript, so hopefully performance shouldn't be hit too badly.

Reply:
Even if it was relatively common, I honestly think this is completely fine as it is.